src/HealthChecks.ApplicationStatus/DependencyInjection/ApplicationStatusHealthCheckBuilderExtensions.cs

```diff
@@ -0,0 +1,98 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+
+namespace HealthChecks.ApplicationStatus.DependencyInjection;
+
+/// <summary>
+/// Extension methods to configure <see cref="ApplicationStatusHealthCheck"/>.
+/// </summary>
+public static class ApplicationStatusHealthCheckBuilderExtensions
+{
+    private const string NAME = "applicationstatus";
+
+    /// <summary>
+    /// Add a health check for Application Status.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="lifetime">Represents an object for intercepting application lifetime events.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'applicationstatus' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <returns>The specified <paramref name="builder"/>.</returns>
+    public static IHealthChecksBuilder AddApplicationStatus(
+        this IHealthChecksBuilder builder,
+        IHostApplicationLifetime lifetime,
+        string? name = default,
+        HealthStatus? failureStatus = default,
+        IEnumerable<string>? tags = default,
+        TimeSpan? timeout = default)
+    {
+        if (lifetime == null)
+        {
+            throw new ArgumentNullException(nameof(IHostApplicationLifetime));
+        }
+
+        builder.Services
+            .TryAddSingleton(sp => new ApplicationStatusHealthCheck(lifetime));
```

Comment:
I see no HC that use `TryAdd`. Why so? Imagine that your app (potentially very big) calls `AddApplicationStatus` without arguments in one place and then calls `AddApplicationStatus` with specifying custom `IHostApplicationLifetime lifetime`. Now you will simply ignore second registration.

Reply:
Intention was to limit possible subscribers to `OnStopping` event. I agree with you but at the same time application can have 10+ registration of `ApplicationStatusHealthcheck` with all that subscriptions, but only one will be used (latest registered)

Reply:
Since now your HC implements IDisposable you don't need to register it.

Reply:
@OlegOLK I wrote some nonsense here. See #1638